### PR TITLE
refactor: RFC 334 Add NIST P and X25519 keys concrete examples 

### PR DESCRIPTION
**Pull request by Baha-sk**
Signed-off-by: Baha Shaaban <baha.shaaban@securekey.com>

### CODE REVIEW

**Review by llorllale — APPROVED** [review]

**Review by kdenhartog — CHANGES_REQUESTED** [review]
Looking through this I'm left a bit confused about the forms being described. For much of the DIDCommV1 stuff I didn't think we had moved to a standards compliant JWE for this work, but I haven't been tracking it for quite some time.

For the DIDCommV2 work most of these key types are supported, but many of the examples are presenting a set of use cases (compact for single recipient, using ECDH-ES to name two particular cases) that aren't defined in the didcomm-messaging spec. 

What's the intent of description for this topic? Is it focused on DIDCommV1 pack/unpack style or DIDCommV2 stuff?

**Comment by llorllale** [comment]
@kdenhartog 

> What's the intent of description for this topic? Is it focused on DIDCommV1 pack/unpack style or DIDCommV2 stuff?

The intent is to advance RFC 0334 for use in DIDComm V1 with an eye on AIP 2.0.

This has nothing to do with DIDComm V2.

**Comment by kdenhartog** [comment]
> @kdenhartog
> 
> > What's the intent of description for this topic? Is it focused on DIDCommV1 pack/unpack style or DIDCommV2 stuff?
> 
> The intent is to advance RFC 0334 for use in DIDComm V1 with an eye on AIP 2.0.
> 
> This has nothing to do with DIDComm V2.

Ahh interesting that may leave us with 3 potential formats that have to be considered at a DIDComm endpoint then while looking to migrate DIDCommV1 to DIDCommV2? E.g. RFC 0019 (old JWE derived pack/unpack), RFC 0334 (Includes Anoncrypt/Authcrypt with similar key types), and the DIDCommV2 version (albeit DIDCommV2 is just a subset of RFC 0334 from the looks of it)

I think this is going to make AIP 2.0 effectively compliant at the envelope layer at least, but probably still going to need mime type adjustments and such to indicate the change in version.

**Review by kdenhartog — COMMENTED** [review]
Changing review to comment now that I understand purpose better. I think I need to do some more reading on AIP 2.0 to consider the transition path to DIDCommV2 (e.g. probably transition to AIP 2.0 to AIP 2.1/AIP 3.0)

File: features/0334-jwe-envelope/anoncrypt-examples.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+      "epk": {
+        "use": "enc",
+        "kty": "EC",
+        "crv": "P-256",
+        "x": "ciiz13A7zt5Ncu7Ouyi1wV29jpU9gQ3UHj0oLGRt_4k",
+        "y": "Ls7zcgkUajNnPp4blZL16OrboTntpHZp74NUScH_Ejw"
+      }
+    },
+    "encrypted_key": "QQoN74Pi06nyX3FR8-raNyQXku0Qx0klg2qo4hzQtN7nY9hKjJxZbg"
+  }],
```
[the remark this answers is not in the document]
Comment (Baha-sk) [reply features/0334-jwe-envelope/anoncrypt-examples.md:62]: currently our Ares-Framework-Go JWE packers don't expose aad directly, but I can probably add aad in these examples (in a future PR though)

[the remark this answers is not in the document]
Comment (Baha-sk) [reply features/0334-jwe-envelope/anoncrypt-examples.md]: I ended up adding aad in this pr @kdenhartog

**Comment by Baha-sk** [comment]
@kdenhartog @andrewwhitehead, I've updated this PR to include integrity checks as recommended by @kdenhartog's [comment](https://github.com/hyperledger/aries-rfcs/pull/575#discussion_r542930798) above.

The examples now include both X25519 and NIST P curved keys.
I've also used DIDComm V2's media types in the examples to be compatible.

This pr is ready for review/merge

**Comment by swcurran** [comment]
Merged per discussion on Aries Working Group call.

**Pull request merged**